### PR TITLE
GrainTracker Parallel Performance Enhancements

### PR DESCRIPTION
**Pull request by permcody**
Spread out joining work - See #11804 for a detailed explanation.

### CODE REVIEW

File: modules/phase_field/src/postprocessors/FeatureFloodCount.C [hunk 1]
```diff
@@ -330,32 +331,138 @@ FeatureFloodCount::communicateAndMerge()
...
+       * global feature maps for each variable.
+       */
+      _communicator.gather_packed_range(i,
+                                        (void *)(nullptr),
+                                        send_buffers.begin(),
+                                        send_buffers.end(),
+                                        std::back_inserter(recv_buffers));
+
+      /**
+       * A call to gather_packed_range seems to populate the receiving buffer on all processors, not
```
Comment (permcody) [inline modules/phase_field/src/postprocessors/FeatureFloodCount.C]: @roystgnr - There's a bug here in gather_packed_range somewhere. I get garbage in `recv_buffers` on processors that "don't" receive so I have to clean it up if I plan to reuse those buffers. I just noticed that my comment is a little misleading. I'm definitely not getting data on all processors (e.g. allgather), but the buffer isn't empty either. I'll see if I can work up a test case or something.

File: modules/phase_field/src/postprocessors/FeatureFloodCount.C [hunk 2]
```diff
@@ -946,12 +1083,15 @@ FeatureFloodCount::mergeSets()
     _partial_feature_sets[map_num].clear();
   }
 
+  //  _console << "\n\n\n\n**********Fully Merged Features*************" << std::endl;
+  //  for (const auto & feature : _feature_sets)
+  //    _console << feature;
+
```
Comment (fdkong) [inline modules/phase_field/src/postprocessors/FeatureFloodCount.C]: Why keep these commented out codes?

File: modules/phase_field/src/postprocessors/GrainTrackerInterface.C [hunk 3]
```diff
@@ -33,6 +33,11 @@ validParams<GrainTrackerInterface>()
                         "Terminate with an error if a grain is created "
                         "(does not include initial callback to start simulation)");
 
+  params.addRangeCheckedParam<short>("verbosity_level",
+                                     1,
+                                     "verbosity_level>=0 & verbosity_level<=2",
+                                     "Prints extra information to the console during operation");
+
```
Comment (fdkong) [inline modules/phase_field/src/postprocessors/GrainTrackerInterface.C]: Could you explain what information would be printed for `0, 1, 2`, respectively ?

File: modules/phase_field/src/postprocessors/GrainTracker.C [hunk 4]
```diff
@@ -74,6 +74,7 @@ GrainTracker::GrainTracker(const InputParameters & parameters)
     _poly_ic_uo(parameters.isParamValid("polycrystal_ic_uo")
                     ? &getUserObject<PolycrystalUserObjectBase>("polycrystal_ic_uo")
                     : nullptr),
+    _verbosity_level(getParam<short>("verbosity_level")),
```
Comment (fdkong) [inline modules/phase_field/src/postprocessors/GrainTracker.C]: `_short_`  sometime just does not work. It is better to just use `int`

**Review by fdkong — CHANGES_REQUESTED** [review]

File: modules/phase_field/src/postprocessors/GrainTracker.C (hunk 4 above)
Comment (permcody) [reply modules/phase_field/src/postprocessors/GrainTracker.C]: No, it **always** works. Do you have a counter example? I use `short` in several places in the framework. I'd probably use `char` except those don't print nicely by default 😄

Comment (permcody) [reply modules/phase_field/src/postprocessors/GrainTracker.C]: @fdkong - I just verified that I can override the parameter from the command line with all three values (0, 1, and 2) and it all works fine.

**Review by fdkong — APPROVED** [review]

**Pull request merged**